eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -232,7 +233,7 @@ object NodeAddress {
     host match {
       case _ if host.endsWith(".onion") && host.length == 22 => Tor2(host.dropRight(6), port)
       case _ if host.endsWith(".onion") && host.length == 62 => Tor3(host.dropRight(6), port)
-      case _ => IPAddress(InetAddress.getByName(host), port).get
+      case _ => IPAddress(InetAddress.getByName(host), port)
```

Comment:
Note: as explained in the method comment, this does a name resolution. We could decide to drop that feature and replace by `InetAddresses.forUriString()` (which would ne a nice symmetry to the newly added `InetAddresses.toUriString()`) and would guarantee us that there is no name resolution, but supporting dns-based node uris seems useful.

Reply:
We will indeed need name resolution (see https://github.com/lightning/bolts/pull/911), I've asked @remyers to take a stab at implementing it once this PR is merged.